onnxruntime/core/platform/windows/env.cc

```diff
@@ -69,80 +78,73 @@ class WindowsEnv : public Env {
     return GetCurrentProcessId();
   }
 
-  static common::Status GetFileSizeIfUnknown(const wchar_t* fname, HANDLE hFile, size_t& length) {
-    if (length > 0) return Status::OK();
+  Status GetFileLength(const ORTCHAR_T* file_path, size_t& length) const override {
+    ScopedFileHandle file_handle{CreateFileW(
+        file_path, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL)};
     LARGE_INTEGER filesize;
-    if (!GetFileSizeEx(hFile, &filesize)) {
+    if (!GetFileSizeEx(file_handle.Get(), &filesize)) {
       int err = GetLastError();
-      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "GetFileSizeEx ", ToMBString(fname), " fail, errcode =", err);
+      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "GetFileSizeEx ", ToMBString(file_path), " fail, errcode = ", err);
     }
     if (static_cast<ULONGLONG>(filesize.QuadPart) > std::numeric_limits<size_t>::max()) {
-      return common::Status(common::ONNXRUNTIME, common::FAIL, "ReadFileAsString: File is too large");
+      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "GetFileLength: File is too large");
```

Comment:
Indeed ORT_MAKE_STATUS is less efficient than the old code, because it needs to construct a ostringstream first.   That's why this line is different with the previous if block. But I'm fine with your change. :-)